iroh-sync/src/sync.rs

```diff
@@ -1377,6 +1409,8 @@ mod tests {
     ) -> Result<()> {
         let alice_peer_id = [1u8; 32];
         let bob_peer_id = [2u8; 32];
+        let mut alice_state = SyncOutcome::default();
+        let mut bob_state = SyncOutcome::default();
```

Comment:
should test that the outcome is the expected

Reply:
:+1:  Added to always verify that num_sent matches num_recv from the other peer. Also added an explicit check to one of the tests, `test_replica_sync`. Could do more, but maybe in a followup.